src/legacy/core_plugins/kbn_vislib_vis_types/public/pie.js

```diff
@@ -50,19 +50,32 @@ export default function HistogramVisType(Private) {
     },
     editorConfig: {
       collections: {
-        legendPositions: [{
-          value: 'left',
-          text: 'left',
-        }, {
-          value: 'right',
-          text: 'right',
-        }, {
-          value: 'top',
-          text: 'top',
-        }, {
-          value: 'bottom',
-          text: 'bottom',
-        }],
+        legendPositions: [
+          {
+            text: i18n.translate('kbnVislibVisTypes.pie.editorConfig.collections.legendPositions.left', {
```

Comment:
Please add a message type at the end of all message ids. 
Also I'd get rid of `collections` segment since I think it's redundant here from the area prospective.

```suggestion
            text: i18n.translate('kbnVislibVisTypes.pie.editorConfig.legendPositions.leftText', {
```

Here is [Guideline](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/GUIDELINE.md#naming-convention) about naming convention.